privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java

```diff
@@ -20,88 +21,78 @@ public PrivacyIDEAAuthenticator() {}
     @Override
     protected final void doExecute(@Nonnull ProfileRequestContext profileRequestContext, @Nonnull PIContext piContext, @Nonnull PIServerConfigContext piServerConfigContext)
     {
-        HttpServletRequest request = this.getHttpServletRequest();
+        HttpServletRequest request = Objects.requireNonNull(this.getHttpServletRequestSupplier()).get();

```

Comment:
I'm not really sure if an exception should be thrown if `HttpServletRequestSupplier` is null. Is is required for authentication to work, if i just want to enter my otp?

Reply:
All form parameters, like otp from the input field, are also transported this way.